File: C#/StreamView/StreamView/View/IStreamView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StreamView.View
{
    interface IStreamView
    {
        #region Properties
        #endregion

        #region Methods

        /// <summary>
        /// Event handler for when the user presses the "next" button.
        /// </summary>
        void nextPressed( object sender, EventArgs e );

        /// <summary>
        /// Event handler for when the user presses the "previous" button.
        /// </summary>
        void prevPressed( object sender, EventArgs e );

        /// <summary>
        /// Event handler for when the user presses the "goto" button.
        /// </summary>
        void gotoPressed( object sender, EventArgs e );

        /// <summary>
        /// Highlights a selection in the current block of data.
        /// </summary>
        /// <param name="bitOffset"></param>
        /// <param name="bitLength"></param>
        void highlightSelection( int bitOffset, int bitLength );

        /// <summary>
        /// Opens a file for viewing.
        /// </summary>
        /// <param name="filename"></param>
        void openFile( string filename );

        #endregion
    }
}

File: Java/StreamView/src/view/IStreamView.java
package view;

import java.io.File;

/*******************************************************************************
 * Simple interface for navigating a file stream.
 ******************************************************************************/
public interface IStreamView
{
    /***************************************************************************
     * Displays the next block of data.
     **************************************************************************/
    void nextPressed();

    /***************************************************************************
     * Displays the previous block of data.
     **************************************************************************/
    void prevPressed();

    /***************************************************************************
     * Displays an individual block of data.
     **************************************************************************/
    void gotoPressed();

    /***************************************************************************
     * Highlights a selection in the block of data.
     * 
     * @param bitOffset
     * @param bitLength
     **************************************************************************/
    void highlightSelection( int bitOffset, int bitLength );

    /***************************************************************************
     * Sets the input file for the underlying data.
     * 
     * @param f
     **************************************************************************/
    void openFile( File f );
}
